### ISSUE TITLE
🔄 Implement deterministic seed management for reproducibility

### ISSUE THREAD

**Issue by mihiarc** [mihiarc/gridfia#6]
## Problem
The package lacks seed management for random operations, making results non-deterministic:
- No global seed setting mechanism
- Parallel processing introduces randomness
- Bootstrap/permutation tests not reproducible
- Results vary between runs

### 1. Global Seed Manager
```python
# bigmap/core/reproducibility.py
import random
import numpy as np
from typing import Optional
from contextlib import contextmanager

class SeedManager:
    """Manage random seeds for reproducibility."""
    
    _global_seed: Optional[int] = None
    
    @classmethod
    def set_global_seed(cls, seed: int):
        """Set global seed for all random operations."""
        cls._global_seed = seed
        random.seed(seed)
        np.random.seed(seed)
        
        # Set seeds for other libraries if present
        try:
            import torch
            torch.manual_seed(seed)
        except ImportError:
            pass
    
    @classmethod
    @contextmanager
    def temporary_seed(cls, seed: int):
        """Context manager for temporary seed."""
        old_random_state = random.getstate()
        old_numpy_state = np.random.get_state()
        
        try:
            random.seed(seed)
            np.random.seed(seed)
            yield
        finally:
            random.setstate(old_random_state)
            np.random.set_state(old_numpy_state)
    
    @classmethod
    def get_seed(cls) -> Optional[int]:
        """Get current global seed."""
        return cls._global_seed
```

### 2. Integration in API
```python
# bigmap/api.py
class BigMapAPI:
    def __init__(self, config=None, seed: Optional[int] = None):
        if seed is not None:
            SeedManager.set_global_seed(seed)
        # ... rest of init
    
    def set_seed(self, seed: int):
        """Set seed for reproducibility."""
        SeedManager.set_global_seed(seed)
```

### 3. Calculation-level seeds
```python
# bigmap/core/calculations/base.py
class ForestCalculation:
    def __init__(self, seed: Optional[int] = None):
        self.seed = seed or SeedManager.get_seed()
    
    def calculate(self, data):
        if self.seed:
            with SeedManager.temporary_seed(self.seed):
                return self._calculate_impl(data)
        return self._calculate_impl(data)
```

### 4. Parallel processing determinism
```python
# bigmap/utils/parallel_processing.py
def process_parallel(func, data, n_jobs=4, seed=None):
    """Deterministic parallel processing."""
    if seed:
        # Create deterministic seeds for workers
        worker_seeds = [seed + i for i in range(n_jobs)]
        results = Parallel(n_jobs=n_jobs)(
            delayed(func)(chunk, seed=s) 
            for chunk, s in zip(data, worker_seeds)
        )
    else:
        results = Parallel(n_jobs=n_jobs)(
            delayed(func)(chunk) for chunk in data
        )
    return results
```

## Acceptance Criteria
- [ ] Global seed manager implemented
- [ ] API supports seed parameter
- [ ] All calculations respect seed setting
- [ ] Parallel processing is deterministic
- [ ] Context manager for temporary seeds
- [ ] Tests verify reproducibility
- [ ] Documentation on reproducibility

## Test Cases
```python
def test_reproducibility():
    # Run with same seed twice
    api1 = BigMapAPI(seed=42)
    result1 = api1.calculate_metrics(data)
    
    api2 = BigMapAPI(seed=42)
    result2 = api2.calculate_metrics(data)
    
    assert np.allclose(result1, result2)

def test_different_seeds():
    api1 = BigMapAPI(seed=42)
    result1 = api1.calculate_metrics(data)
    
    api2 = BigMapAPI(seed=123)
    result2 = api2.calculate_metrics(data)
    
    assert not np.allclose(result1, result2)
```

## References
- Data Science Review: Reproducibility section
- NumPy random seed documentation
- scikit-learn reproducibility guidelines
